### PR TITLE
Change to use MQTT and Websockets

### PR DESCRIPTION
**Pull request by schan-solace**

### CODE REVIEW

File: cardemo/src/main/resources/static/js/build.js [hunk 1]
```diff
@@ -236,7 +250,62 @@ var MyGame = {};
         this.helpButton.inputEnabled = true;
       }, this);
       
-      this.apiClient = new APIclient();
+      var carId = generateID();
+
+        var hostname = "tcp.apps.pcfdemo.solacemessaging.net";
```
Comment (jschabowsky) [inline cardemo/src/main/resources/static/js/build.js]: We really need to figure out a way to make this more portable such that we dont have to change the code each time the tcp routes change.

File: cardemo/src/main/resources/static/js/build.js [hunk 2]
```diff
@@ -236,7 +250,62 @@ var MyGame = {};
...
+
+        // connect the MQTT client
+        client.connect(options);
+        console.log('Connecting...');
+
+        // called when the client connects
+        function onConnect(context) {
+          // Once a connection has been made, make a subscription and send a message.
+          console.log("Connected");
+          //client.subscribe("World");
```
Comment (jschabowsky) [inline cardemo/src/main/resources/static/js/build.js]: Delete these commented out lines

File: carmap/src/main/resources/static/TopicSubscriber.js [hunk 3]
```diff
@@ -327,30 +361,29 @@ var TopicSubscriber = function (topicName) {
...
         } else {
             //var host = document.getElementById('host').value;
             //var vpn = document.getElementById('vpn').value;
             //var user = document.getElementById('user').value;
             //var pw = document.getElementById('password').value;
-            var host = "10.244.0.3:7000";
-            var vpn = "v003";
-            var user = "v003.cu000017";
-            var pw = "16698a80-0381-422d-a69e-aa6fb30eb273";
+            var host = "tcp.apps.pcfdemo.solacemessaging.net:16322";
```
Comment (jschabowsky) [inline carmap/src/main/resources/static/TopicSubscriber.js]: Same comment as above, need a better way to do this.

File: carmap/src/main/resources/static/TopicSubscriber.js [hunk 4]
```diff
@@ -210,6 +242,7 @@ var TopicSubscriber = function (topicName) {
         infoContent += '</p>';
         cars[id]["info"].setContent(infoContent);
         //setTimeout(function(){ marker.setAnimation(null); },100);
```
Comment (jschabowsky) [inline carmap/src/main/resources/static/TopicSubscriber.js]: remove this line.

**Review by jschabowsky — CHANGES_REQUESTED** [review]
In general looks really good, Just clean up lines which are commented out to make it cleaner if a client were to be reading through the code base.

**Comment by schan-solace** [comment]
Done.

Do I need to submit a new pull request, or is the existing one still active?

Steve

**Comment by mdspielman** [comment]
I have asked @jschabowsky for a final review.

**Comment by jschabowsky** [comment]
Looks good to me. Lets merge as I will need this baseline for spring one as well!

**Comment by mdspielman** [comment]
Thanks @jschabowsky. Merging.

**Pull request merged**